File: src/main/java/com/strange/safety/inquiry/entity/Inquiry.java
package com.strange.safety.inquiry.entity;

import com.strange.safety.common.entity.BaseEntity;
import com.strange.safety.user.entity.User;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "inquiries")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Inquiry extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inquiry_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private InquiryCategory category;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private InquiryStatus status;

    @Column(name = "reply_content", columnDefinition = "TEXT")
    private String replyContent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "replied_by")
    private User repliedBy;

    @Column(name = "replied_at")
    private LocalDateTime repliedAt;

    @Builder
    private Inquiry(User user, InquiryCategory category, String title, String content) {
        this.user = user;
        this.category = category;
        this.title = title;
        this.content = content;
        this.status = InquiryStatus.WAITING;
    }

    public void addAnswer(String replyContent, User admin) {
        this.replyContent = replyContent;
        this.repliedBy = admin;
        this.repliedAt = LocalDateTime.now();
        this.status = InquiryStatus.COMPLETED;
    }

    public boolean isAnswered() {
        return this.status == InquiryStatus.COMPLETED;
    }
}

File: src/shared/types/inquiry.ts
export type InquiryCategory = '카메라 및 영상' | '알림 및 경보' | '모바일' | '기타';
export type InquiryCategoryRaw = 'CAMERA_VIDEO' | 'ALERT_ALARM' | 'MOBILE' | 'OTHER';

export interface Inquiry {
  id: number;
  userEmail: string;
  userName: string;
  userRole: 'INDIVIDUAL' | 'CORPORATE';
  category: InquiryCategory;
  title: string;
  content: string;
  status: 'WAITING' | 'COMPLETED';
  replyContent: string | null;
  repliedByName: string | null;
  repliedAt: string | null;
  createdAt: string;
}
